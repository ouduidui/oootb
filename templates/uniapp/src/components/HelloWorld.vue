<script setup lang="ts">
import { nav } from 'router/index'
const title = ref('Hello')

const toAbout = () => nav('about')
</script>

<template>
  <view
    class="font-momo text-center font-light p-5 flex-col items-center justify-center"
  >
    <view>HelloWorld</view>
    <button class="w-20 text-sm p-1 mt-4" @tap="toAbout">
      About
    </button>
  </view>
</template>
