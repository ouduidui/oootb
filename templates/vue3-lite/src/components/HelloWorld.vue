<script setup lang="ts">
const count = ref(0)

const addHandler = () => count.value++
</script>

<template>
  <div>
    <p text op75>Count: {{ count }}</p>
    <button m-3 text-sm btn bg-blue @click="addHandler">Add</button>
  </div>
</template>

<style scoped></style>
