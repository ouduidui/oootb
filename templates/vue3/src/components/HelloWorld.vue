<script setup lang="ts">
const props = defineProps<{
  initial: number
}>()

const count = ref(props.initial)

const addHandler = () => count.value++
</script>

<template>
  <div>
    <p text op75>
      Count: {{ count }}
    </p>
    <button m-3 text-sm btn bg-blue class="inc" @click="addHandler">
      Add
    </button>
  </div>
</template>
