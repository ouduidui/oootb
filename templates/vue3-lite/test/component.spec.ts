import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import HelloWorld from '../src/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('should render', () => {
    const wrapper = mount(HelloWorld, { props: { initial: 10 } })
    expect(wrapper.text()).toContain('10')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async() => {
    const wrapper = mount(HelloWorld, { props: { initial: 0 } })
    expect(wrapper.text()).toContain('0')

    expect(wrapper.find('.inc').exists()).toBe(true)

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')
  })
})
