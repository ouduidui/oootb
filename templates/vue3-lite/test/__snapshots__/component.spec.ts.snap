// Vitest Snapshot v1

exports[`HelloWorld.vue > should render 1`] = `
"<div>
  <p text=\\"\\" op75=\\"\\"> Count: 10</p><button m-3=\\"\\" text-sm=\\"\\" btn=\\"\\" bg-blue=\\"\\" class=\\"inc\\"> Add </button>
</div>"
`;
